Core/include/Acts/Geometry/PortalLinkBase.hpp

```diff
@@ -0,0 +1,64 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Utilities/BinningType.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <memory>
+
+namespace Acts {
+
+class RegularSurface;
+class TrackingVolume;
+class GeometryContext;
+
+class PortalLinkBase {
+ public:
+  PortalLinkBase(std::shared_ptr<RegularSurface> surface)
+      : m_surface(std::move(surface)) {}
+
+  virtual ~PortalLinkBase() = default;
+
+  // @TODO: Does this need boundary tolerance?
```

Comment:
potentially an "on surface" tolerance would be good

Reply:
And pipe it through the relevant global to local calls and vice versa?

Reply:
Yes - in general I feel like the parameter should not be defaulted so people have to actively think what to use and why it might be relevant for them. But that is another story